doc/pages/Manual.md

```diff
@@ -7,33 +7,228 @@ primarily of use for packagers, package maintainers and repository maintainers.
   comprehensive built-in documentation `opam [command] --help`.
 * For a gentler introduction to packaging, see the
   [Packaging guide](Packaging.html)
-* For a more complete specification, and if you are interested in more details
-  on the layout of `~/.opam`, you may want to check the
-  [OPAM Developer's Manual](http://opam.ocaml.org/doc/manual/dev-manual.html)
-  (OUTDATED)
 * If you want to hack on opam or build related tools, the API documentation can
   be browsed [here](api/index.html)
 
+## Base concepts
+
+### Opam root
+
+Opam holds its configuration, metadata, logs, temporary directories and caches
+within a directory that we will call _opam root_. By default, this is `~/.opam`,
+and we may refer to it by this name in this manual for the sake of simplicity,
+but this can be changed using the `OPAMROOT` environment variable or the
+`--root` command-line argument.
+
+An existing opam root is required for opam to operate normally, and one is
+created upon running `opam init`. The initial configuration can be defined
+through a configuration file at `~/.opamrc`, `/etc/opam` or at a location
+specified through the `--config` command-line option. If none is present, opam
+defaults to its built-in configuration that binds to the OCaml repository at
+`https://opam.ocaml.org`.
+
+Except in explicit cases, opam only alters files within its opam root. It is
+organised as follows:
+- [`~/.opam/config`](#config): the global opam configuration file
+- `~/.opam/repo/`: contains the mirrors of the configured package repositories
+- [`~/.opam/repo/repos-config`](#reposconfig): lists the configured package repositories and their URLs
+- `~/.opam/repo/<repo>`: mirror of the given repository
+- `~/.opam/opam-init/`: contains opam configuration scripts for the outside world, e.g. shell environment initialisation
+- `~/.opam/cache/`: caches of downloaded files
+- `~/.opam/plugins/<plugin>`: reserved for plugins
+- `~/.opam/<switch>`: prefixes of named [switches](#switches)
+
+### Repositories
+
+Repositories are collection of opam package definitions. They respect the
+following hierarchy:
+- [`/repo`](#repo): repository configuration file
+- [`/packages/<pkgname>/<pkgname>.<version>/opam`](#opam): holds the metadata
+  for the given package. `url` and `descr` may also be present, in which case
+  they override what may already be present in the `opam` file
```

Comment:
Indeed, that is for compatibility; the choice was made originally so that pinning could still work by adding an `url` file to override the package source, but it no longer works that way (pinning is now just an indirection of the package's definition, including url).
Either way could be really confusing though ; having both could also be an error, at least on the developer side. I can change the way it works.